runway/data_types.py

```diff
@@ -324,7 +330,7 @@ def __init__(self, name='category', description=None, choices=None, default=None
     def deserialize(self, value):
         if value not in self.choices:
             msg = 'category value "%s" does not appear in choices list.' % value
-            raise InvalidArgumentError(self.name, msg)
+            raise InvalidArgumentError(self.type, msg)
         return value
```

Comment:
Name isn't guaranteed to be non-`None` so I chose to use `type` here instead. But maybe we should only use `type` if name is `None`.

Reply:
I imagine an error of type `number: Invalid argument` will be unclear, so I'd vote for `self.name` or self.type`.

Reply:
Or, to avoid repeating that pattern across types, passing the entire data type instance to the `InvalidArgumentError` constructor.

Reply:
I chose to repeat the pattern over changing the signature of the `InvalidArgumentError`, keeping it general.